### ISSUE TITLE
[Bug] ttnn.i1 clamps its input at 88.5 and returns one constant for 48% of the bfloat16 domain

### ISSUE THREAD

**Issue by kinginu** [tenstorrent/tt-metal#52174]
**Component / Area:** kernels

**Issue Type:** Bug

## Observed

`ttnn.i1` clamps its input to `±88.5` and evaluates there, so above that bound it is a **constant function**:

```
ttnn.i1(89) == ttnn.i1(1000) == ttnn.i1(3e38) == 1.15477e+37
```

**31132 of the 65024 normal bfloat16 patterns, 47.9% of the domain, return that one value** — one distinct absolute output across the whole region. The result is finite, carries no `Inf` and no `NaN`, and nothing downstream can detect it. `ttnn.i1(±Inf)` also returns `±1.15e37`.

Twelve of those patterns have a true value that is finite and representable in bfloat16, and are wrong by 1.6x to 20x:

| x | `ttnn.i1(x)` | true i1(x), float64 | device / true |
|---|---|---|---|
| 88.5 | 1.15477e+37 | 1.14987e+37 | 1.004 |
| 89 | 1.15477e+37 | 1.89053e+37 | 0.611 |
| 90 | 1.15477e+37 | 5.11061e+37 | 0.226 |
| 91.5 | 1.15477e+37 | 2.27172e+38 | **0.0508** |
| 1000 | 1.15477e+37 | +Inf | - |

The remaining 31120 patterns have a true value of `+Inf` and receive a finite `1.15e37`.

## The clamp bound was taken from the wrong function

`ckernel_sfpu_i1.h:95` reads `// Clamp to [-88.5, 88.5] — exp() saturates near ±88.7 in FP32`. That is true of `exp`, but not of `i1`. The kernel computes `i1(x) = exp(x)/sqrt(2·pi·x) · P(1/x)`, and the `sqrt(2·pi·x)` divisor is about 23.6 in this region:

| quantity | overflows fp32 at |
|---|---|
| `exp(x)` | 88.72284 |
| **`i1(x)`** | **91.90626** |

`ln(sqrt(2*pi*88.5)) = 3.16` is the whole of the gap. The clamp exists to protect the bare `exp(|x|)` intermediate formed at `:52-54`, and in doing so it discards 3.16 of domain on which `i1` itself is finite and representable.

## Expected

Outside the fit domain an op should return the correct value, or saturate to `+Inf` where the true value overflows the output format. It should not return the value at the clamp point, because that converts an out-of-range input into a plausible in-range answer that nothing downstream can detect.

The distinction that matters is whether the true function is genuinely flat past the bound. `erf` is clamped at `|x| = 10` and that is correct, because `erf` really does saturate to 1. `i1` diverges.

## Steps to reproduce

```python
import torch, ttnn
d = ttnn.open_device(device_id=0)
x = torch.zeros((1, 1, 32, 32), dtype=torch.bfloat16)
for i, v in enumerate([88.5, 91.5, 1000.0, 3e38]):
    x[0, 0, 0, i] = v
t = ttnn.from_torch(x, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=d)
print(ttnn.to_torch(ttnn.i1(t))[0, 0, 0, :4])
# tensor([1.1548e+37, 1.1548e+37, 1.1548e+37, 1.1548e+37], dtype=torch.bfloat16)
# true:   1.1499e+37, 2.2717e+38, inf,        inf
ttnn.close_device(d)
```

## Why no test catches it

A test exists and **encodes the clamp into its own golden**. `tests/ttnn/unit_tests/operations/eltwise/test_unary_i1.py:154`, `test_i1_clamp_boundary`:

```python
# i1 is unbounded; out-of-clamp inputs (|x| > 88.5) return i1(±88.5).
expected_input = torch.clamp(boundaries, min=-88.5, max=88.5)
torch_output_tensor = torch.special.i1(expected_input)
```

The reference is computed from the clamped input, so the test asserts that the device reproduces the clamp rather than that it computes `i1`. It passes today and pins the defect in place: the 88.5 is hardcoded in the golden, so any fix makes this test fail and it has to be rewritten as part of the change. `sfpu_domains.py:325` draws `I1` from `UNIFORM(-3.75, 3.75)`, which does not reach even the `|x| > 10` asymptotic branch.

## Scope

- `tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_i1.h:95` — `x = sfpi::symmetric_clamp(x, I1_MAX_INPUT);` with `I1_MAX_INPUT = 88.5f`.

The file is byte-identical under `tt_metal/hw/ckernels/wormhole_b0/`, so this is not Blackhole-only.

## What fixing it costs

Measured, in the attached fix: the BF16 math kernel's `.text` is unchanged at 1200 bytes and the FP32 one grows by one instruction, 1376 to 1380 bytes, with 27 other math kernels byte-identical. Wall clock at `[16, 8, 192, 384]` is within run-to-run noise on both dtypes.

## The same clamp is currently being propagated into `i0`

The open `ttnn.i0` large-|x| accuracy issue holds `i1` up as the reference implementation and instructs implementers to clamp near the exp overflow bound "exactly as i1 does". Both of its open fix PRs, `#52126` and `#51843`, add `I0_MAX_INPUT = 88.5f`. `i0` in fact stays finite in fp32 to 91.90076. Whichever merges, `ttnn.i0` acquires a constant output over roughly 48% of the bfloat16 domain, on a binding that documents no range restriction.

## This is one instance of a class

The same mechanism — a saturating clamp on the input that turns the op into a constant outside the fit domain — is already filed separately in five sibling kernels: `erfc` `#51137`, `digamma` `#51128`, `snake_beta` `#51116`, `sin`/`cos` `#51845`, `softplus` `#51866`. `i1` is the instance nobody has reported, because it is the one the others are told to copy.

## Environment

- Blackhole P300, Ubuntu 22.04, tt-metal `d8ab5b156a`, built Release.
- Reference `torch.special.i1` in float64, torch 2.11.0+cpu.
- Exhaustive over all 65536 bfloat16 patterns rather than sampled; denormals, denormal results, `±0`, `Inf` and `NaN` classified and counted separately rather than filtered.

## Not established

- `i1` is bound with `"[Validated range: -10 to 10]"` (`unary_nanobind.cpp:1741`), so a strict reading puts everything above outside what the op promises. Against that: the kernel implements an explicit asymptotic branch for `|x| > 10` at `:135`, so it deliberately serves inputs far outside the documented range, and `i0`, which is about to inherit the same clamp, carries no range note at all.
- Wormhole is inferred from byte-identical sources, not measured. There is no Wormhole hardware on this host.
- `Bfp8_b` was not run. It should behave the same, since the clamp is in the math kernel and the format only affects pack.
